opts/opts.go

```diff
@@ -292,6 +293,77 @@ func (o *NamedMapOpts) Name() string {
 	return o.name
 }
 
+// SetOpts holds a map of values and a validation function.
+type SetOpts struct {
+	values map[string]bool
+}
+
+// Set validates if needed the input value and add it to the
+// internal map, by splitting on '='.
+func (opts *SetOpts) Set(value string) error {
+	k, v, found := strings.Cut(value, "=")
+	var isSet bool
+	if !found {
+		isSet = true
+		k = value
+	} else {
+		var err error
+		isSet, err = strconv.ParseBool(v)
+		if err != nil {
+			return err
+		}
+	}
+	opts.values[k] = isSet
+	return nil
+}
+
+// GetAll returns the values of MapOpts as a map.
+func (opts *SetOpts) GetAll() map[string]bool {
+	return opts.values
+}
+
+func (opts *SetOpts) String() string {
+	return fmt.Sprintf("%v", opts.values)
+}
```

Comment:
Slightly wondering if this should produce a string based on the `map[string]bool` or an output based on the (more human-readable) comma-separated list.

Do you know where the string-format surfaces? Also; is the stringer interface required for FlagOpts, or would this be something we could postpone (add when needed?)